src/hooks/useGetData.jsx

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from 'react';
+
+export default function useGetData(url) {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState();
+  const [data, setData] = useState([]);
+
+  const getData = async () => {
+    try {
+      setLoading(true);
+      setError(undefined);
```

Comment:
```suggestion
  const [error, setError] = useState('');
  const [data, setData] = useState([]);

  const getData = async () => {
    try {
      setLoading(true);
```

에러 state에 담길 데이터의 타입이 string이라면 기본값을 정해주는 것이 좋겠네요!